core/src/mindustry/world/blocks/power/SeamlessSolarGenerator.java

```diff
@@ -0,0 +1,66 @@
+package mindustry.world.blocks.power;
+
+import arc.*;
+import arc.graphics.g2d.*;
+import mindustry.content.*;
+import mindustry.world.*;
+
+import static mindustry.Vars.tilesize;
+
+public class SeamlessSolarGenerator extends SolarGenerator{
+    private TextureRegion[] compass = new TextureRegion[8];
+
+    public SeamlessSolarGenerator(String name){
+        super(name);
+    }
+
+    @Override
+    public void load(){
+        super.load();
+
+        for(int i = 0; i < compass.length; i++){
+            compass[i] = Core.atlas.find(name + "-" + i);
+        }
+    }
+
+    @Override
+    protected TextureRegion[] generateIcons(){
+        return new TextureRegion[]{Core.atlas.find(name + "-full")};
+    }
+
+    @Override
```

Comment:
yes, i am aware the code below here is not dry and could be improved by using e.g. vectors/blendbits/etc 🙂